core/src/main/java/com/opensymphony/xwork2/factory/DefaultConverterFactory.java

```diff
@@ -42,7 +42,7 @@ public void setContainer(Container container) {
 
     public TypeConverter buildConverter(Class<? extends TypeConverter> converterClass, Map<String, Object> extraContext) throws Exception {
         LOG.debug("Creating converter of type [{}]", converterClass.getCanonicalName());
-        return container.getInstance(converterClass);
+        return container.inject(converterClass);
```

Comment:
why?

Reply:
`getInstance()` can only create instances of classes already registered in container, thus means you would have to register all the converters used in your app as beans in `struts.xml`.

`inject()` is a factory method, it will create an instance of the class and inject all the marked dependencies.

`buildConverter()` was not used (a dead code), as `buildBean` was used instead to create converters, that's why that wasn't discovered early.